plugins/lenovo-thinklmi/fu-lenovo-thinklmi-plugin.c

```diff
@@ -83,6 +84,38 @@ fu_lenovo_thinklmi_plugin_device_registered(FuPlugin *plugin, FuDevice *device)
 	}
 }
 
+static void
+fu_lenovo_thinklmi_plugin_add_security_attrs(FuPlugin *plugin, FuSecurityAttrs *attrs)
+{
+	FwupdBiosSetting *bios_attr;
+	FuContext *ctx = fu_plugin_get_context(plugin);
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+
+	/* check if boot order lock is turned on */
+	bios_attr = fu_context_get_bios_setting(ctx, SECURE_ROLLBACK);
+	if (bios_attr == NULL) {
+		g_debug("failed to find %s in cache\n", SECURE_ROLLBACK);
+		return;
+	}
+
+	/* create attr */
+	attr = fu_plugin_security_attr_new(plugin, FWUPD_SECURITY_ATTR_ID_UEFI_ROLLBACK_PROTECTION);
+	fu_security_attr_add_bios_target_value(attr, SECURE_ROLLBACK, "enable");
+
+	fu_security_attrs_append(attrs, attr);
+
+	if (g_strcmp0(fwupd_bios_setting_get_current_value(bios_attr), "Disable") == 0) {
+		fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_ACTION_CONFIG_FW);
+		fwupd_security_attr_set_result(attr, FWUPD_SECURITY_ATTR_RESULT_NOT_ENABLED);
+		return;
+	}
+
+	/* success */
+	fwupd_security_attr_set_result(attr, FWUPD_SECURITY_ATTR_RESULT_ENABLED);
+	fwupd_security_attr_add_obsolete(attr, "pci_psp");
```

Comment:
Wanted to discuss this specifically...

Given the AMD solution is hardware based but the OEM ones might not be; I was thinking maybe the OEM ones maybe we should put at a lower level, and **not obsolete** the AMD one.  They're mutually exclusive.  You can have the OEM protection and the AMD one simultaneously.

This has an implication that on AMD you can't get to HSI 4 unless you have the AMD one at least then, but you can still get the OEM ones turned on for lower levels (whatever level we decide)

Reply:
(I've mocked up the PR accordingly)